vizro-core/src/vizro/models/_components/form/range_slider.py

```diff
@@ -60,23 +60,30 @@ def build(self):
             Output(self.id, "value"),
             Output(f"temp-store-range_slider-{self.id}", "data"),
         ]
-        input = [
+        inputs = [
             Input(f"{self.id}_start_value", "value"),
             Input(f"{self.id}_end_value", "value"),
             Input(self.id, "value"),
             State(f"temp-store-range_slider-{self.id}", "data"),
+            State(f"{self.id}_callback_data", "data"),
         ]
 
-        @callback(output=output, inputs=input)
-        def update_slider_values_callback(start, end, slider, input_store):
-            trigger_id = callback_context.triggered_id
-
-            return self._update_slider_values(
-                start_txt=start, end_txt=end, slider=slider, input_store=input_store, value=value, trigger_id=trigger_id
-            )
+        clientside_callback(
+            ClientsideFunction(namespace="clientside", function_name="update_range_slider_values"),
+            output=output,
+            inputs=inputs,
+        )
 
         return html.Div(
             [
+                dcc.Store(
+                    f"{self.id}_callback_data",
+                    data={
+                        "id": self.id,
+                        "min": self.min,
+                        "max": self.max,
+                    },
+                ),
```

Comment:
(for later) Why is this now needed?

Reply:
When the RangeSlider callback was implemented as a server-side callback, it could use global variables from the class instance such as: self.id, self.min, self.max...

Since the current solution is implemented as a client-side javascript callback, this means that the callback cannot access global or class instance variables. This means that we had to save this information on the client side (hide it in dcc.Store).